tpm2/src/util/ski.c

```diff
@@ -0,0 +1,560 @@
+/**
+ * tpm2_kmyth_ski.c:
+ *
+ * C library containing utilities related to the ski format for
+ * Kmyth applications using TPM 2.0.
+ */
+#include "tpm2_kmyth_seal.h"
+#include "tpm2_kmyth_global.h"
+#include "kmyth_cipher.h"
+#include "tpm2_kmyth_misc.h"
+#include "tpm2_kmyth_session.h"
+#include "tpm2_kmyth_io.h"
+#include "tpm2_kmyth_key.h"
+#include "tpm2_pcrManagement.h"
+#include "tpm2_kmyth_object.h"
+#include "tpm2_config_tools.h"
+#include "tpm2_info_tools.h"
+#include "ski.h"
+#include "tpm2_kmyth_mu.h"
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include <openssl/rand.h>
+
+//############################################################################
+// tpm2_kmyth_parse_ski_bytes
+//############################################################################
+int tpm2_kmyth_parse_ski_bytes(uint8_t * input, size_t input_length,
```

Comment:
It really doesn't thrill me that we're modifying `input` during function calls within this function. I get why it's happening, but I think it would be clearer if we used a separate `position` pointer that gets modified.

Reply:
I think the use of an offset parameter (rather than modifying the input) is a good suggestion.